### PR TITLE
Update markdown test dependencies

### PR DESCRIPTION
**Pull request by RDIL**
# PR Summary
Bumps some packages used by the static analysis to the latest version

Bumps the following:

`"markdownlint": "^0.11.0"` -> `"markdownlint": "^0.12.0"`,
`"through2": "^2.0.1"` -> `"through2": "^3.0.1"`

`devDependencies`:
`"gulp-debug": "^3.0.0"` -> `"gulp-debug": "^4.0.0"`

## PR Context
Dependencies out of date

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by RDIL** [comment]
Requesting review: @JamesWTruher @markekraus

**Comment by RDIL** [comment]
CC @adityapatwardhan

**Comment by RDIL** [comment]
cc @TravisEz13

File: test/common/markdown/package-lock.json [hunk 1]
```diff
@@ -25,6 +25,15 @@
       "resolved": "https://registry.npmjs.org/ansi-regex/-/ansi-regex-2.1.1.tgz",
       "integrity": "sha1-w7M6te42DYbg5ijwRorn7yfWVN8="
     },
+    "ansi-styles": {
```
Comment (iSazonov) [inline test/common/markdown/package-lock.json]: Do we really need the file package-lock.json?

**Review by iSazonov — COMMENTED** [review]
@TravisEz13 Can DependaBot monitor this and update?

File: test/common/markdown/package-lock.json (hunk 1 above)
Comment (RDIL) [reply test/common/markdown/package-lock.json]: Yes, it speeds up installing

Comment (RDIL) [reply test/common/markdown/package-lock.json]: Also dependabot an only do 1 language

**Comment by RDIL** [comment]
@TravisEz13 please review

**Comment by RDIL** [comment]
@iSazonov I do know a bot that I think can do it by the name of [Greenkeeper](https://greenkeeper.io)

**Comment by RDIL** [comment]
@iSazonov I've done some research and I know how to set up greenkeeper to bump these dependencies automatically if you want me to set it up.

**Comment by iSazonov** [comment]
@RDIL Thanks! We use depedabot and I don't know does it make sense to add one more.

**Comment by RDIL** [comment]
@iSazonov sadly dependabot can only do one language

**Comment by iSazonov** [comment]
We could consider to switch to greenkeeper.

**Comment by RDIL** [comment]
@isazonov the problem with that is greenkeeper is JavaScript ONLY and dependabot can only do ONE language.  We will need to use both for it to work.

File: test/common/markdown/package-lock.json (hunk 1 above)
Comment (TravisEz13) [reply test/common/markdown/package-lock.json]: dependabot does actually support multiple languages.  I added this folder and we will see the results.

**Comment by TravisEz13** [comment]
dependabot definitely supports multiple languages.

File: test/common/markdown/package-lock.json (hunk 1 above)
Comment (RDIL) [reply test/common/markdown/package-lock.json]: Got it.  If it tries to bump the folder I will approve it's PRs.

**Comment by TravisEz13** [comment]
Adding a new extension is expensive process for us.  Let us see if we can get dependabot to work first.

**Comment by TravisEz13** [comment]
dependabot has opened #9092, #9093 and #9094

**Comment by RDIL** [comment]
Dependabot is stealing my commits 😡

**Comment by TravisEz13** [comment]
lol

**Comment by RDIL** [comment]
this tyranny must be stopped

File: test/common/markdown/package-lock.json (hunk 1 above)
Comment (TravisEz13) [reply test/common/markdown/package-lock.json]: Also, our compliance team wants them

File: test/common/markdown/package-lock.json [hunk 2]
```diff
@@ -978,6 +998,17 @@
       "requires": {
         "graceful-fs": "^4.1.11",
         "through2": "^2.0.3"
```
Comment (TravisEz13) [inline test/common/markdown/package-lock.json]: should this be `^2.0.5`?

Comment (RDIL) [reply test/common/markdown/package-lock.json]: Not sure, but my bot did it and it looks like it works.

**Review by TravisEz13 — APPROVED** [review]
nothing blocking

File: test/common/markdown/package-lock.json (hunk 2 above)
Comment (TravisEz13) [reply test/common/markdown/package-lock.json]: Not blocking

**Comment by TravisEz13** [comment]
@iSazonov 	if you are okay with this PR, Can you merge it and close the equivalent PRs dependabot opened?

**Pull request merged**